src/runtime/contrib/tensorrt/tensorrt_builder.cc

```diff
@@ -0,0 +1,421 @@
+}
+
+void TensorRTBuilder::VisitExpr_(const VarNode* node) {
+  const int id = TrackVarNode(node);
+
+  const std::string& tensor_name = node->name_hint();
+  auto shape = GetShape(node->checked_type());
+  // Remove batch dim
+  if (shape.size() > 1) shape.erase(shape.begin());
+  DLOG(INFO) << "Added TRT network input: " << node->name_hint() << " "
+             << DebugString(shape);
+  nvinfer1::Dims dims = VectorToTrtDims(shape);
+  auto type_node = node->checked_type().as<TensorTypeNode>();
+  CHECK(type_node != nullptr &&
+        runtime::TypeMatch(type_node->dtype, kDLFloat, 32))
+      << "Only FP32 inputs are supported.";
+  auto input =
+      network_->addInput(tensor_name.c_str(), nvinfer1::DataType::kFLOAT, dims);
+  network_input_map_[id] = tensor_name;
+  node_output_map_[node] = {TrtOpInput(input)};
+}
+
+void TensorRTBuilder::VisitExpr_(const ConstantNode* node) {
+  nvinfer1::Weights weight = GetNdArrayAsWeights(node->data, kDLCPU);
+  auto shape = node->data.Shape();
+  // Remove batch dim.
+  if (shape.size() > 1 && shape[0] == 1) shape.erase(shape.begin());
+  nvinfer1::Dims dims = VectorToTrtDims(shape);
+  auto const_layer = network_->addConstant(dims, weight);
+  CHECK(const_layer != nullptr);
+  node_output_map_[node] = {TrtOpInput(const_layer->getOutput(0))};
+}
+
+void TensorRTBuilder::VisitExpr_(const CallNode* call) {
+  AddTrtLayerParams params(network_, call, &trt_weights_);
+  // Look up converter.
+  auto it = GetOpConverters()->find(params.op_name);
+  CHECK(it != GetOpConverters()->end())
+      << "Unsupported operator conversion to TRT, op name: " << params.op_name;
+  const auto converter = it->second;
+
+  // Ensure that nodes are processed in topological order by visiting their
+  // inputs first.
+  for (size_t i = 0; i < call->args.size(); ++i) {
+    // Handle special case where input must be constant array on CPU.
+    if (!converter->variable_input_count &&
+        converter->input_types[i] == kWeight) {
+      // Input must be a constant weight
+      if (auto* var = call->args[i].as<VarNode>()) {
+        GetInputAsWeights(var);
+      } else if (auto* node = call->args[i].as<ConstantNode>()) {
+        GetConstantAsWeights(node);
+      } else {
+        // Temporary workaround for transposed weights. Once partitioning is
+        // available, the transpose will be computed by tvm and the result will
+        // be a var input.
+        if (auto* transpose = call->args[i].as<CallNode>()) {
+          if (transpose->op.as<OpNode>()->name == "transpose") {
+            if (auto* weights = transpose->args[0].as<VarNode>()) {
+              GetInputAsTransposedWeights(transpose, weights);
+            } else {
+              LOG(FATAL) << "TRT requires a constant input here.";
+            }
+          } else {
+            LOG(FATAL) << "TRT requires a constant input here.";
+          }
+        } else {
+          LOG(FATAL) << "TRT requires a constant input here.";
+        }
```

Comment:
```
        {
        CallNode* transpose = call->args[i].as<CallNode>();
        VarNode* weights;
        if (transpose != nullptr
            && transpose->op.as<OpNode>()->name == "transpose"
            && (weights = transpose->args[0].as<VarNode>()) != nullptr) {
          GetInputAsTransposedWeights(transpose, weights);
        } else {
          LOG(FATAL) << "TRT requires a constant input here."
        }
        }
```

Reply:
Thanks, I was wondering how to clean that up.